packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -209,6 +209,7 @@ abstract class RenderSliverScrollingPersistentHeader extends RenderSliverPersist
     final double paintExtent = maxExtent - constraints.scrollOffset;
     geometry = new SliverGeometry(
       scrollExtent: maxExtent,
+      paintOrigin: constraints.overlap,
```

Comment:
You might want to do this only when the overlap is negative.  Consider the case when you have a pinned header followed by this kind of header.  You want this kind of header to scroll under the pinned header.  The way you've written this, this header will shift its paint location by the overlap between the two.